orttraining/orttraining/python/training/_ortmodule_graph_execution_manager.py

```diff
@@ -30,28 +30,39 @@ def _run_forward(execution_session, onnx_model, device, *inputs, **kwargs):
 
     # TODO: Try to reuse the output buffers as some of the output tensors are same sizes,
     #   especially the backward graph outputs.
-    io_binding = execution_session.io_binding()
-    run_options = C.RunOptions()
-
-    # Use IO binding
-    _utils._create_iobinding(io_binding, inputs, onnx_model, device)
-
-    # Run and return module outputs.
-    ort_output = execution_session.run_forward(io_binding, run_options)
-    forward_outputs, run_id = ort_output.ortvalues, ort_output.run_id
-
-    user_outputs = tuple(_utils._ortvalue_to_torch_tensor(forward_output) for forward_output in forward_outputs)
+    # REVIEW(codemzs): Consolidate Training Agent with InferenceAgent on C++ side to not
+    # have the need for passing IOBinding.
```

Comment:
would there be a need for partial run in no_grad mode too? OK to be in a separate PR but I'd like to see IOBinding being cleared away from ORTModule in general. It is meant to separate the data movement to device, from the computation on device. Since PyTorch explicitly moves data to device before inference, there's no need to use IOBindings.

Reply:
I agree.